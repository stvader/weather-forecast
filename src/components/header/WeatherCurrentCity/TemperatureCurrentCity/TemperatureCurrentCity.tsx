import React from 'react';
import { Typography } from 'antd';

const { Title } = Typography;

const TemperatureCurrentCity: React.FC<any> = ({ temperature }: any) => (
  <Title level={3}>{temperature}</Title>
);

export default TemperatureCurrentCity;
