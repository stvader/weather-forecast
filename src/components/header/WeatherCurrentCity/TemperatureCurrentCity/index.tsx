import React from 'react';
import { connect } from 'react-redux';

import { IAppState } from '../../../../reducers/types';
import { selectTempCurrent } from '../../../../selectors';
import TemperatureCurrentCity from './TemperatureCurrentCity';
import { getTemperatureView, ITemperatureView } from '../../../../utils/getTemperatureView';

interface ITemperatureCurrentCity {
  temperatureData: ITemperatureView;
}

const TemperatureCurrentCityContainer: React.FC<ITemperatureCurrentCity> = ({
  temperatureData,
}: ITemperatureCurrentCity) => {
  const temperature = getTemperatureView(temperatureData);

  return <TemperatureCurrentCity temperature={temperature} />;
};

const mapStateToProps = (state: IAppState) => ({
  temperatureData: selectTempCurrent(state),
});

export default connect(mapStateToProps)(TemperatureCurrentCityContainer);
