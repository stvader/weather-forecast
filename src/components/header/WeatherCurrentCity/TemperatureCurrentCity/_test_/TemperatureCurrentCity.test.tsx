import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Typography } from 'antd';
import TemperatureCurrentCity from '../TemperatureCurrentCity';

const { Title } = Typography;

configure({ adapter: new Adapter() });

describe('TemperatureCurrentCity', () => {
  it('render TemperatureCurrentCity', () => {
    const component = shallow(<TemperatureCurrentCity />);

    const title = component.find(Title);

    expect(component).toMatchSnapshot();
    expect(title).toBeDefined();
  });
});
