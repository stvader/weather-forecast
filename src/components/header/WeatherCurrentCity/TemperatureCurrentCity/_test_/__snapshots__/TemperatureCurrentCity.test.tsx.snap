// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureCurrentCity render TemperatureCurrentCity 1`] = `ShallowWrapper {}`;
